### PR TITLE
[REVIEW] Add assert_eq and test reductions and binary operators

### LINKED ISSUE #520: Test against Pandas with heavily parametrized tests?
I spent a bit of time to look through the cudf testing suite.  It's pretty slick!

It tends to switch between semi-exhaustive parametrized tests and one-off tests for certain functionality.  *If* we choose to mostly follow Pandas semantics, then I'd like for us to consider embracing parametrization more.  I suspect that we might be able to establish good coverage of the Pandas API by writing tests like the following:

```python
@pytest.fixture
def df():
    return pd.DataFrame(...)

@pytest.fixture
def gf(df):
    return cudf.DataFrame.from_pandas(df)

@pytest.mark.parametrize(...)
@pytest.mark.parametrize(...)
def test_class_of_operations(df, gf, ...):
     assert_eq(func(df), func(gf))
```

This is similar in principle to existing code like the following:

https://github.com/rapidsai/cudf/blob/7aa3799e73948899a85e70bc73d3cea50b65dff1/python/cudf/tests/test_binops.py#L114-L122

I'm suggesting that we just take this a bit further.

### PR DESCRIPTION
**Pull request by mrocklin**
This adds an `assert_eq` function for testing and then shows off the use of that function while testing reductions and binary operators.  This serves as a proof of concept for testing generally as described in 
https://github.com/rapidsai/cudf/issues/520

I think that this PR can stand on its own, but we should wait for a decision in #520 .  My apologies for pushing something before discussion.

Closes #520

### CODE REVIEW

**Comment by kkraus14** [comment]
@mrocklin can you mark this as [WIP] or [REVIEW] so we know when you're done iterating? A lot of people like to iterate on PRs and take advantage of the CI system 😄

**Comment by mrocklin** [comment]
Sure.  I've marked as a WIP.  I'll change to REVIEW once I get CI to pass

File: python/cudf/tests/test_dataframe.py [hunk 1]
```diff
@@ -1001,3 +1003,62 @@ def test_dataframe_shape_empty():
     gdf = DataFrame()
 
     assert pdf.shape == gdf.shape
+
+
+@pytest.fixture
+def df():
+    return pd.DataFrame({'x': range(10),
+                         'y': range(10)})
```
Comment (thomcom) [inline python/cudf/tests/test_dataframe.py]: In the existing tests `pdf` is used really consistently for pandas dataframes.

Comment (mrocklin) [reply python/cudf/tests/test_dataframe.py]: Fixed.  I've also changed the gf pronoun to gdf to match (hopefully this is welcome as well)

**Pull request merged**